API.md

```diff
@@ -1,12 +1,120 @@
 # Authmosphere API Documentation
 
+## Token Cache
+
+Class to request and cache tokens on client-side.
+
+#### Usage
+
+```typescript
+
+// TokenCacheOptions
+const cacheOptions = {
+  cacheConfig: {
+    percentageLeft: 0.75
+  },
+  logger: someExternalLogger
+};
+
+const oAuthConfig = {
+  grantType: OAuthGrantType.CLIENT_CREDENTIALS_GRANT,
+  accessTokenEndpoint: 'https://....',
+  credentialsDir: './credentialsDir'
+};
+
+const tokenConfig = {
+  'service-foo': ['foo.read', 'foo.write'],
+  'service-bar': ['bar.read']
+};
+
+// create a new cache
+const tokenCache = new TokenCache(tokenConfig, oAuthConfig, cacheOptions);
+
+// Get a token from cache
+tokenCache
+  .get('service-foo') // Needs to match with a key from 'tokenConfig'
+  .then((token: Token) => {
+    console.log(token.access_token);
+  });
+```
+
+### constructor
+
+#### Signature
+
+```ts
+constructor(tokenConfig, oauthConfig: TokenCacheOAuthConfig, options)
+```
+
+#### Arguments
+
+* `tokenConfig: [key: string]: string[]` - the token cache is able to handle multiple tokens. The string array should contain the scopes attached to the token.
+* `oauthConfig: TokenCacheOAuthConfig` -
+  Either [`ClientCredentialsGrantConfig`](#clientcredentialsgrantconfig) or [`PasswordCredentialsGrantConfig`](#passwordcredentialsgrantconfig)` plus the additional `tokenInfoEndpoint: string` property that specifies the URL of the token validation endpoint.
+* [`options?: TokenCacheOptions`](./src/types/TokenCacheConfig.ts)
+  * [`cacheConfig?: CacheConfig`](./src/types/TokenCacheConfig.ts)
+    * `percentageLeft: number` - To determine when a token is expired locally (means when to issue a new token): if the token exists for `((1 - percentageLeft) * lifetime)` then issue a new one.
+  * [`logger?: Logger`](#logging)
+
+### get
+
+Returns cached token or requests a new one if lifetime as configured in `cacheOptions.cacheConfig` is expired.
```

Comment:
"Returns cached token or requests a new one if lifetime (as configured in `cacheOptions.cacheConfig`) is expired." (brackets)

Reply:
✅